notebooks/cugraph_benchmarks/release.ipynb

```diff
@@ -0,0 +1,628 @@
+    "    \n",
+    "    # Calculating modularity scores for comparison \n",
+    "    _ = community.modularity(parts, _G)  \n",
+    "    \n",
+    "    t2 = time.time() - t1\n",
+    "    return t2\n",
+    "\n",
+    "def cu_louvain(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_cu_ugraph(_df)\n",
+    "    _,_ = cugraph.louvain(_G)\n",
+    "    t2 = time.time() - t1\n",
+    "    return t2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Triangle Counting"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def nx_tc(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_nx_ugraph(_df)\n",
+    "    nx_count = nx.triangles(_G)\n",
+    "    \n",
+    "    # To get the number of triangles, we would need to loop through the array and add up each count\n",
+    "    count = 0\n",
+    "    for key, value in nx_count.items():\n",
+    "        count = count + value    \n",
+    "    \n",
+    "    t2 = time.time() - t1\n",
+    "    return t2\n",
+    "\n",
+    "def cu_tc(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_cu_ugraph(_df)\n",
+    "    _ = cugraph.triangles(_G)\n",
+    "    t2 = time.time() - t1\n",
+    "    return t2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Benchmark Functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# number of datasets\n",
+    "num_datasets = len(data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# do a simple pass just to get all the libraries initiallized\n",
+    "v = './data/preferentialAttachment.mtx'\n",
+    "gdf = read_data(v)\n",
+    "print(f\"\\tGDF Size {len(gdf)}\")\n",
+    "\n",
+    "g = create_cu_ugraph(gdf)\n",
+    "\n",
+    "print(f\"\\tcugraph Size {g.number_of_edges()}\")\n",
+    "print(f\"\\tcugraph Order {g.number_of_vertices()}\")\n",
+    "\n",
+    "del gdf\n",
+    "del g\n",
+    "gc.collect()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# arrays to capture performance gains\n",
+    "names = []\n",
+    "\n",
+    "# Two dimension data\n",
+    "time_algo_cu = []       # will be two dimensional\n",
+    "time_algo_nx = []       # will be two dimensional\n",
+    "perf = []\n",
+    "\n",
+    "\n",
+    "\n",
+    "i = 0\n",
+    "for k,v in data.items():\n",
+    "    time_algo_cu.append([])\n",
+    "    time_algo_nx.append([])\n",
+    "    perf.append([])\n",
```

Comment:
You could also just create a new list for each at the start of the loop, then just append them at the bottom. That would eliminate the need to have `i` and these appends of `[]`, and might read a little better:
```
cuGraphTimes = []
nxTimes = []
speedups = []
...
tx = nx_bc(pdf)
tc = cu_bc(gdf)
nxTimes.append(tx)
cuGraphTimes.append(tc)
speedups.append(tx/tc)
...
time_algo_nx.append(nxTimes)
time_algo_cu.append(cuGraphTimes)
perf.append(speedups)
```

Reply:
True.